### ISSUE TITLE
"bad argument #2 to '?' (out of range at /home/miruware/sdc/linh/torch/pkg/torch/generic/Tensor.c:913)" Error

### ISSUE THREAD

**Issue by thunderizik** [mks0601/V2V-PoseNet_RELEASE#47]
Hi I've been trying to train and test on NYU hand dataset and though I followed the instructions including changing data dir, convert png to bin, etc... but still getting this same error:

> Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5db: NYU mode: train
training data loading...
testing data loading...
invalid frame in test set: 0
Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruwle CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/milib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Enviruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Founle CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruwinvalid frame in test set: 0
Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/mirch/pkg/torch/generic/Tensor.c:913)ruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5==> training startFound Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5Found Environment variable CUDNN_PATH = /home/miruware/anaconda3/envs/linh/lib/libcudnn.so.5epoch: 0/10 batch: 40000/72757 loss: 0.00031782130122301
==> testing:
/home/miruware/sdc/linh/torch/install/bin/luajit: bad argument #2 to '?' (out of range at /home/miruware/sdc/linh/torch/pkg/torch/generic/Tensor.c:913)
stack traceback:                                                                                                                                                                                                 "digitbox" 12:46 25-Apr-1
        [C]: at 0x7f4574773b30
        [C]: in function '__index'
        test.lua:28: in function 'test'
        run_me.lua:58: in main chunk
        [C]: in function 'dofile'        ...linh/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
        [C]: at 0x00405e90

Do you have any ideas what is going on? And what should I do?
Thanks in advance.

**Comment by mks0601** [comment]
That errors comes from test.lua:28, which is getting reference point. Did you check you load reference point properly?

**Comment by thunderizik** [comment]
I just checked and I think there's nothing wrong since the directories for dataset and centers are all right. And since loading the reference point of testing phase is almost the same while training, and I got through training phase so i think i should be the same with testing phase, imo

**Comment by mks0601** [comment]
I think you'd better debug from data loading part

**Comment by thunderizik** [comment]
Hi,
Thanks for your quick reply! I try to debug and i found out that in test.lua at this part:

`--generate batch
        for i = t,t+curBatchNum-1 do
            
            local input_name = testName[i]
            local refPt = testRefPt[i]:clone()
            local depthimage = load_depthmap(input_name)

            refPts[i-t+1] = refPt
            
            --canceling data augmentation (it was prepared before)
            newSz = 100
            angle = 0
            local trans = torch.Tensor(worldDim)
            trans[1] = originalSz/2 - croppedSz/2 + 1
            trans[2] = originalSz/2 - croppedSz/2 + 1
            trans[3] = originalSz/2 - croppedSz/2 + 1
            
            --voxelizing
            inputs[i-t+1] = generate_cubic_input(inputs[i-t+1][1],depthimage,refPts[i-t+1],newSz,angle,trans)
            
        end`
when I try to print(i) then the loop goes up to 8253 and gave that error mentioned in the topic which exceeds the range (which is only 8252 for test set of NYU) so maybe that is reason. Any suggestions?

**Comment by thunderizik** [comment]
Ok I found the problem. Thanks alot! 👍

**Comment by ngalym** [comment]
Hi @thunderizik , I am getting the same error. Can you tell how you resolved the issue? thank you

**Comment by chienerh** [comment]
I also encountered this error. I changed [line 25 in src/test.lua](https://github.com/mks0601/V2V-PoseNet_RELEASE/blob/b91a69f3031869a192ccaf6b7339a3e4c5fb7eb7/src/test.lua#L25) to `for i = t,t+curBatchNum-2 do` and avoided the error message. I'm not sure if this is the best way to solve this.

**Comment by chienerh** [comment]
I also encountered this problem. I found out it is because the batch size is not an aliquot part of the test data size.

**Comment by mks0601** [comment]
Sorry for late reply. I think you should check 'i' of the 'test.py' at line 25 is outside of the size of the  'testRefPt' at line 28. In my case, I haven't encountered this problem for all datasets.

**Comment by chienerh** [comment]
Yeah, that was my problem. If `batchSz` is not a factor of `testDataSz`, the `i` will exceed the size of `testRefPt` for the last batch on line 28. Thanks for replying though!
